screencastr.thor

```diff
@@ -54,16 +47,27 @@ class Screencastr < Thor
     pad=#{options[:width]}:#{options[:height]}:(ow-iw)/2:(oh-ih)/2' #{out_file}`
   end
 
-  desc "concat FIRST_IN SECOND_IN OUT_FILE", "Concatenate two video files together"
-  def concat(first_in, second_in, out_file)
-    `ffmpeg -i #{first_in} -i #{second_in} \
-    -filter_complex '[0:v] scale=#{options[:width]}:#{options[:height]}:force_original_aspect_ratio=decrease,\
-    pad=#{options[:width]}:#{options[:height]}:(ow-iw)/2:(oh-ih)/2 [vs0]; \
-    [1:v] scale=#{options[:width]}:#{options[:height]}:force_original_aspect_ratio=decrease,\
-    pad=#{options[:width]}:#{options[:height]}:(ow-iw)/2:(oh-ih)/2 [vs1]; \
-    [vs0][0:a][vs1][1:a] concat=n=2:v=1:a=1 [vout][aout]' \
-    -r #{options[:framerate]} \
-    -map '[vout]' -map '[aout]' #{out_file}`
+  desc "concat FIRST_IN SECOND_IN ... NTH_IN OUT_FILE", "Concatenate an arbitrary number of files together"
+  def concat(*in_files, out_file)
+    if in_files.length <= 1
+      puts "concat requires at least 2 inputs"
+      exit
+    end
+
+    inputs = []
+    scale_filters = []
+    streams = []
+
+    in_files.each_with_index do |in_file, index|
+      inputs << "-i #{in_file}"
+      scale_filters << "[#{index}:v] scale=#{options[:width]}:#{options[:height]}:force_original_aspect_ratio=decrease,\
+      pad=#{options[:width]}:#{options[:height]}:(ow-iw)/2:(oh-ih)/2 [vs#{index}];"
+      streams << "[vs#{index}][#{index}:a]"
+    end
+
+    `ffmpeg #{inputs.join(" ")} -filter_complex '#{scale_filters.join(" ")}\
+    #{streams.join} concat=n=#{streams.length}:v=1:a=1 [vout][aout]' \
+    -r #{options[:framerate]} -map '[vout]' -map '[aout]' #{out_file}`
```

Comment:
What's the output of running this system command? Would be good if there's some progress reported while this stuff is going on. Probably a separate task if it's not all dumped to `stdout` already.

Reply:
It's dumped to `stdout`, its the same as just running `ffmpeg` in terminal. Gives you a decent progress report of what's happening.